### ISSUE TITLE
Rule Change: `no-useless-computed-key` should support ES2022 class fields

### ISSUE THREAD

**Issue by sosukesuzuki** [eslint/eslint#16326]
### What rule do you want to change?

`no-useless-computed-key` should support ES2022 class fields

### What change to do you want to make?

Generate more warnings

### How do you think the change should be implemented?

A new default behavior

### Example code

```js
/*eslint no-useless-computed-key: "error"*/
class Foo {
  ["a"] = "a";
}
```

### What does the rule currently do for this code?

Doesn't report any errors for computed keys with class fields.

### What will the rule do after it's changed?

Reports errors for computed keys with class fields.

### Participation

- [X] I am willing to submit a pull request to implement this change.

### Additional comments

Class fields: https://github.com/tc39/proposal-class-fields

**Comment by sosukesuzuki** [comment]
Ah thank you, sorry I didn't know.
